code/modules/mob/living/silicon/ai/life.dm

```diff
@@ -1,171 +1,181 @@
+
+/mob/living/silicon/ai/proc/life_handle_unpowered_core()
+	overlay_fullscreen("blind", /obj/abstract/screen/fullscreen/blind)
+	if(client)
+		change_sight(removing = SEE_TURFS|SEE_MOBS|SEE_OBJS)
+		see_in_dark = 0
+	see_invisible = SEE_INVISIBLE_LIVING
+
+	if (aiRestorePowerRoutine != 0)
+		return
+	aiRestorePowerRoutine = 1
+	to_chat(src, "You've lost power!")
+	spawn(2 SECONDS)
+		if(!aiRestorePowerRoutine)
+			return // Checking for premature changes.
+		to_chat(src, "Backup battery online. Scanners, camera, and radio interface offline. Beginning fault-detection.")
+		sleep(5 SECONDS)
+		if(!aiRestorePowerRoutine)
+			return // Checking for premature changes.
+		if(is_ai_powered())
+			power_restored()
+			return
+		to_chat(src, "Fault confirmed: missing external power. Shutting down main control system to save power.")
+		sleep(2 SECONDS)
+		if(!aiRestorePowerRoutine)
+			return // Checking for premature changes.
+		to_chat(src, "Emergency control system online. Verifying connection to power network.")
+		sleep(5 SECONDS)
+		if(!aiRestorePowerRoutine)
+			return // Checking for premature changes.
+		if (istype(get_turf(src), /turf/space))
+			to_chat(src, "Unable to verify! No power connection detected!")
+			aiRestorePowerRoutine = 2
+			return
+		to_chat(src, "Connection verified. Searching for APC in power network.")
+		sleep(5 SECONDS)
+		if(!aiRestorePowerRoutine)
+			return // Checking for premature changes.
+		var/obj/machinery/power/apc/theAPC = null
+
+		var/PRP //like ERP with the code, at least this stuff is no more 4x sametext
+		for (PRP=1, PRP<=4, PRP++)
+			if(!aiRestorePowerRoutine)
+				return // Checking for premature changes.
+			var/area/AIarea = get_area(src)
+			for (var/obj/machinery/power/apc/APC in AIarea)
+				if (!(APC.stat & BROKEN))
+					theAPC = APC
+					break
+			if (!theAPC)
+				switch(PRP)
+					if (1)
+						to_chat(src, "Unable to locate APC!")
+					else
+						to_chat(src, "Lost connection with the APC!")
+				aiRestorePowerRoutine = 2
+				return
+			if (is_ai_powered())
+				power_restored()
+				return
+			switch(PRP)
+				if (1)
+					to_chat(src, "APC located. Optimizing route to APC to avoid needless power waste.")
+				if (2)
+					to_chat(src, "Best route identified. Hacking offline APC power port.")
+				if (3)
+					to_chat(src, "Power port upload access confirmed. Loading control program into APC power port software.")
+				if (4)
+					to_chat(src, "Transfer complete. Forcing APC to execute program.")
+					sleep(50)
+					if(!aiRestorePowerRoutine)
+						theAPC = null
+						return // Checking for premature changes.
+					to_chat(src, "Receiving control information from APC.")
+					sleep(2)
+					if(!aiRestorePowerRoutine)
+						theAPC = null
+						return // Checking for premature changes.
+					//bring up APC dialog
+					theAPC.attack_ai(src)
+					aiRestorePowerRoutine = 3
+					to_chat(src, "Here are your current laws:")
+					show_laws()
+			spawn(5 SECONDS)
+			theAPC = null
```

Comment:
possible indentation issue, otherwise i don't know what's going on here

Reply:
That's how it was before I touched it, I'm still not quite sure how that entire proc works.

Reply:
this spawn, effectively, does nothing
spawn only affects code indented under it

Reply:
guess I'll remove it